backend/modules/publishing/service.py

```
@@ -1 +1,587 @@
+                setattr(simulation, 'case_study_url', s3_url)
+                self.db.add(simulation)
+                logger.info(f"[PDF_STORAGE] Uploaded PDF to S3: {s3_url}")
+            return
+        
+        # Upload from temp URL
+        if temp_pdf_url:
+            # Download from temp location and upload to final location
+            # Extract S3 key from temp URL if it's already in S3
+            if _is_s3_url(temp_pdf_url):
+                # Extract key from URL
+                parts = temp_pdf_url.split('/')
```

Ruff (0.14.8)

Local variable `parts` is assigned to but never used

Remove assignment to unused variable `parts`

(F841)